packages/data/addon/utils/classes/cache.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2022, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+
+type CacheRecord<T> = {
+  creationTime: number; // the time this record was created
+  lastAccessed: number; // the last time this record was accessed
+  records: T; // the actual records
+};
+
+/**
+ * a cache
+ * to use, give it a unique string (cacheId) and whatever you want stored alongside that key
+ * returns only exact matches
+ */
+export default class Cache<T> {
+  private _maxCacheSize;
+  private _cacheTimeout;
+  private _cache: Record<string, CacheRecord<T>> = {};
+  getCacheId;
+
+  /**
+   * @param {function (params:type) => string} cacheIdFunction - a function that returns unique strings associated with specific items in cache
+   * @param {number} maxCacheSize - the maximum number of items that can be stored in the cache (default 10)
+   * @param {number} cacheTimeout - the maximum amount of milliseconds a record should be allowed to stay in the cache (default 1 hour)
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  constructor(cacheIdFunction: (args: any) => string, maxCacheSize = 10, cacheTimeout = 60 * 60 * 1000) {
+    this.getCacheId = cacheIdFunction;
+    this._maxCacheSize = maxCacheSize;
+    this._cacheTimeout = cacheTimeout;
+  }
+
+  /**
+   * removes items that were created too long ago from the cache
+   */
+  private _removeStaleItems(): void {
```

Comment:
consider returning number of deletes (could be useful for creating cache stats later)